### PR TITLE
Feature/spring boot upgrade and trivy

### PR DESCRIPTION
**Pull request by southeo**
Spring boot and trivy

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -5,7 +5,7 @@
 	<parent>
 		<groupId>org.springframework.boot</groupId>
 		<artifactId>spring-boot-starter-parent</artifactId>
-		<version>2.7.9</version>
+		<version>3.0.6</version>
```
Comment (samleeflang) [inline pom.xml:8]: Can go to 3.1.0

File: pom.xml [hunk 2]
```diff
@@ -103,8 +100,13 @@
 			<version>${json-patch.version}</version>
 		</dependency>
 		<dependency>
-			<groupId>org.keycloak</groupId>
-			<artifactId>keycloak-spring-boot-starter</artifactId>
+			<groupId>org.springframework.boot</groupId>
+			<artifactId>spring-boot-starter-oauth2-resource-server</artifactId>
+		</dependency>
+		<dependency>
```
Comment (samleeflang) [inline pom.xml:106]: Is this needed? If so can we add the version to the properties?

File: src/main/java/eu/dissco/core/digitalmediaobjectprocessor/controller/DigitalMediaObjectController.java [hunk 3]
```diff
@@ -29,7 +29,6 @@ public class DigitalMediaObjectController {
 
   private final ProcessingService processingService;
 
-  @PreAuthorize("isAuthenticated()")
```
Comment (samleeflang) [inline src/main/java/eu/dissco/core/digitalmediaobjectprocessor/controller/DigitalMediaObjectController.java:32]: This call should be authenticated

File: src/main/java/eu/dissco/core/digitalmediaobjectprocessor/security/WebSecurityConfig.java [hunk 4]
```diff
@@ -1,45 +1,31 @@
...
-
+public class WebSecurityConfig {
+  private final JwtAuthConverter jwtAuthConverter;
   @Bean
-  @Override
-  protected SessionAuthenticationStrategy sessionAuthenticationStrategy() {
-    return new NullAuthenticatedSessionStrategy();
+  public SecurityFilterChain securityFilterChain(HttpSecurity http) throws Exception {
+    http.authorizeHttpRequests()
+            .requestMatchers(EndpointRequest.to(HealthEndpoint.class)).permitAll()
```
Comment (samleeflang) [inline src/main/java/eu/dissco/core/digitalmediaobjectprocessor/security/WebSecurityConfig.java:21]: Can we also only expose the /health endpoint? Not sure what we are exposing this way, but looks more than we would like:
https://www.baeldung.com/spring-boot-actuators

File: src/main/resources/application.properties [hunk 5]
```diff
@@ -0,0 +1,3 @@
+spring.security.oauth2.resourceserver.jwt.issuer-uri
+spring.security.oauth2.authorizationserver.endpoint.jwk-set-uri=
```
Comment (samleeflang) [inline src/main/resources/application.properties:2]: Did you check these properties, do they really exists?

**Review by samleeflang — COMMENTED** [review]
Couple of comments

File: pom.xml (hunk 2 above)
Comment (southeo) [reply pom.xml:106]: 1) We need it for the JwtAuthConverter
2) Version is managed by spring boot, it's at 3.1.0 right now

File: src/main/java/eu/dissco/core/digitalmediaobjectprocessor/controller/DigitalMediaObjectController.java (hunk 3 above)
Comment (southeo) [reply src/main/java/eu/dissco/core/digitalmediaobjectprocessor/controller/DigitalMediaObjectController.java:32]: It is authenticated (See WebSecurityConfig). Anything that isn't a GET request or actuator/health has to be authenticated with a JWT token. This annotation is made redundant

File: src/main/java/eu/dissco/core/digitalmediaobjectprocessor/security/WebSecurityConfig.java (hunk 4 above)
Comment (southeo) [reply src/main/java/eu/dissco/core/digitalmediaobjectprocessor/security/WebSecurityConfig.java:21]: There are two steps to the actuator endpoints. 
1. Determine which endpoints are exposed
2. Determine which of the exposed endpoints need to be 

By default, only `/actuator/health` is exposed, so it's okay to put a blanket permission here. If we want to implement more actuator endpoints, we can revisit this. 

FYI You can verify what endpoints are exposed and what they reveal with the swager ui, just set the following in application.properties
`springdoc.show-actuator=true`

File: src/main/resources/application.properties (hunk 5 above)
Comment (southeo) [reply src/main/resources/application.properties:2]: Removed the unnecessary one (the second property)

**Pull request merged**